app/services/ccms/requestors/case_add_requestor.rb

```diff
@@ -168,12 +168,14 @@ def generate_provider_details(xml)
       end
 
       def generate_correspondence_address(xml)
-        xml.__send__(:"common:AddressLine1", applicant.address.address_line_one)
-        xml.__send__(:"common:AddressLine2", applicant.address.address_line_two)
-        xml.__send__(:"common:City", applicant.address.city)
-        xml.__send__(:"common:County", applicant.address.county)
+        xml.__send__(:"common:CoffName", applicant.correspondence_address_for_ccms.care_of_recipient) unless applicant.correspondence_address_for_ccms.care_of_recipient.nil?
+        xml.__send__(:"common:AddressLine1", applicant.correspondence_address_for_ccms.address_line_one)
+        xml.__send__(:"common:AddressLine2", applicant.correspondence_address_for_ccms.address_line_two)
+        xml.__send__(:"common:AddressLine3", applicant.correspondence_address_for_ccms.address_line_three) unless applicant.correspondence_address_for_ccms.address_line_three.nil?
+        xml.__send__(:"common:City", applicant.correspondence_address_for_ccms.city)
+        xml.__send__(:"common:County", applicant.correspondence_address_for_ccms.county)
         xml.__send__(:"common:Country", "GBR")
-        xml.__send__(:"common:PostalCode", applicant.address.pretty_postcode)
+        xml.__send__(:"common:PostalCode", applicant.correspondence_address_for_ccms.pretty_postcode) if applicant.correspondence_address_for_ccms.postcode.present?
```

Comment:
Maybe it's my migraine, however, I presume `post code` is always required, regardless of the correspondence preference, therefore, do we need the conditional statement?

Reply:
It's been a while, but I think this may have been added because the postcode validation is added in the form, not the model.  So test harnesses could have been created with no postcodes and then failed here... maybe

Reply:
👍  not a blocker